src/rsht.c

```diff
@@ -139,12 +139,17 @@ void rsht_destroy(rsht_ht *ht) {
   free(ht->items);
 }
 
-void rsht_foreach(rsht_ht *ht, rsht_callback fn, void *userdata) {
+size_t rsht_foreach(rsht_ht *ht, rsht_callback fn, void *userdata) {
+  size_t num_iterated = 0;
```

Comment:
Why declare a new variable instead of using `i` to keep track?

Reply:
`i` will increase even when the key is `NULL`--I only increment `num_iterated` when `fn` is run and returns true.

Reply:
Ah, true! Would it be possible to keep track of the number of items skipped instead and return the index iterated to minus the number of items skipped? I think that would be easier to follow.

Reply:
This will be inconsistent, since `i` will be incremented if the loop reaches its end, but it will not if we break early because `fn` returned false for one of the items.